BUILDING.md

```diff
@@ -1,47 +1,65 @@
 # Building with CMake
 
+This project encourages uses of up-to-date build tools and contemporary development environments.
+To build from sources, you need:
+
+- CMake 3.26 or higher
+- A C++ compiler that supports C++23 (or equivalent) which is just a bug fix of C++20
+- An implementation of the C++ Core Guidelines Support Library (GSL)
+
 ## Dependencies
 
 For a list of dependencies, please refer to [vcpkg.json](vcpkg.json).
 
+This project depends on the [Microsoft implementation of GSL](https://github.com/microsoft/GSL), which is available on all supported platforms.  We recommend that you use your favorite package manager to install it prior to building and installing the IFC SDK from source.  If you just want to build the project and try it without installing it, then there is no need to install the GSL, but in that case the build step is a bit more involved.
+
+On Windows platforms, we recommend using [vckpg](https://vcpkg.io/en/getting-started.html) as the C++ package manager.  Issue
+```sh
+$ ./vckpg install --triplet x64-windows ms-gsl
+```
+in the VCPKG directory, to install the GSL.  The above command assumes you are building for the `x64` target.  Replace that with the appropriate triplet if you are taregting a different platform like `arm64`.
+
+On linux platforms, there is no commonly agreed-upon name for that package.  Here are a few common name for the Microsoft implementation of the GSL on some linux distributions if you're using the system package manager:
+
+- arch: `microsoft-gsl`
+- debian: `ms-gsl`
+- fedora: `guidelines-support-library`
+- gentoo: `dev-cpp/ms-gsl`
+- ubuntu: `ms-gsl`
+
+Please, consult your linux distribution manual to find the appropriate package name.
+
 ## Build
 
 Here are the steps for building in release mode with a single-configuration
-generator, like the Unix Makefiles one:
+generator, as with Unix Makefiles:
 
 ```sh
 cmake -B build -D CMAKE_BUILD_TYPE=Release
 cmake --build build
 ```
 
 Here are the steps for building in release mode with a multi-configuration
-generator, like the Visual Studio ones:
+generator, as with MSBuild projects:
 
 ```sh
 cmake -B build
 cmake --build build --config Release
 ```
 
-### Building with MSVC
-
-If you are building this project with MSVC, you need to pass some flags (like `/permissive-` and other `/Zc` switches) to requires ISO C++ conformant behavior. See the `flags-windows` preset in the
-[CMakePresets.json](CMakePresets.json) file for the flags and with what
-variable to provide them to CMake during configuration.
```

Comment:
> Does `/std:c++latest` and/or will `/std:c++23` cover these?

What are the "specific these" here?

Reply:
The flags that this paragraph calls the users' attention to. MSVC by default is not ISO C++ conforming and I believe this is an important detail.

Reply:
> The flags that this paragraph calls the users' attention to. MSVC by default is not ISO C++ conforming and I believe this is an important detail.

The instructions here are for building the project.  That paragraph does not add anything because the actual flags are in the `CMakePresets.json`.